### DISCUSSION TITLE
long-run testing service

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#5629] [Ideas]
We should have a long-run testing service that loads data via streaming & mock queries continuously.
Thus we can trust that this version is stable enough to produce.

**Comment by zhang2014** [comment]
👍 We need to add it to the release channels.

**Comment by Xuanwo** [comment]
Github Action has a 6 hours hard limit for a single job.

> Job execution time - Each job in a workflow can run for up to 6 hours of execution time. If a job reaches this limit, the job is terminated and fails to complete.

Is it enough?

references: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration

**Comment by xudong963** [comment]
This long-term testing service should have a feature that once a bug is found, it automatically raises an issue, and then continues to run tests without interruption.
